### PR TITLE
feat: FDN-183 FDN-184 FDN-186 Open Feature Provider and Refactor

### PR DESCRIPTION
**Pull request by JamieSinn**
Test harness is expected to be broken due to the change of signature for the client initialization.

- Changed the signature of the client init to match the rest of the SDKs
- Use php8 typing to properly get valid return values and not "array" blindly.

- Setup more unit tests for provider usage/matching expectations
- Cleanup and refactor the initialization options 
- Cleanup and remove the variable HTTP request path to remove redundant generated code.

### CODE REVIEW

**Comment by JamieSinn** [comment]
Test harness update: https://github.com/DevCycleHQ/test-harness/pull/365

File: examples/hello-world/README.md [hunk 1]
```diff
@@ -3,6 +3,11 @@
 Install dependencies with composer:
 `composer update`
 
+Set the following environment variables:
```
Comment (jonathannorris) [inline examples/hello-world/README.md:6]: How? from where?

File: examples/hello-world/composer.json [hunk 2]
```diff
@@ -2,8 +2,14 @@
...
         "php": "^8.0",
-        "devcycle/php-server-sdk": "1.*"
+        "devcycle/php-server-sdk": "*"
     },
+    "repositories": [{
+        "type": "path",
+        "url": "../../",
+        "only": ["devcycle/php-server-sdk"]
+    }
+    ],
```
Comment (jonathannorris) [inline examples/hello-world/composer.json:12]:
```suggestion
    }],
```

Comment (JamieSinn) [reply examples/hello-world/composer.json:12]: PHP Syntax autoformat.

File: examples/hello-world/README.md (hunk 1 above)
Comment (JamieSinn) [reply examples/hello-world/README.md:6]: I'll elaborate.

File: lib/Api/DevCycleClient.php [hunk 3]
```diff
@@ -1,48 +1,27 @@
...
- * https://openapi-generator.tech
- * Do not edit the class manually.
- */
-
 namespace DevCycle\Api;
 
 use DevCycle\Model\DevCycleEvent;
 use DevCycle\Model\DevCycleUser;
 use DevCycle\Model\DevCycleUserAndEventsBody;
+use DevCycle\Model\ErrorResponse;
```
Comment (jonathannorris) [inline lib/Api/DevCycleClient.php:8]: Probably not needed for this PR, but I think we should try and split off a bunch of this file into smaller parts, maybe moving the request parts?

Comment (JamieSinn) [reply lib/Api/DevCycleClient.php:8]: I tried to start doing that by pulling the code into functions; definitely something we want to do in the long run.

File: lib/Model/DevCycleUser.php [hunk 4]
```diff
@@ -716,8 +702,91 @@ public function __toString()
      *
      * @return string
      */
-    public function toHeaderValue()
+    public function toHeaderValue(): string
     {
         return json_encode(ObjectSerializer::sanitizeForSerialization($this));
     }
+
+    public static function FromEvaluationContext(EvaluationContext $context): DevCycleUser
```
Comment (jonathannorris) [inline lib/Model/DevCycleUser.php:710]: Do we have unit tests for this?

Comment (JamieSinn) [reply lib/Model/DevCycleUser.php:710]: not directly - it's on my list to do

Comment (jonathannorris) [reply lib/Model/DevCycleUser.php:710]: like for all the edge-cases and error handling? 

Do any of these limitation's / restrictions similarly apply here? https://docs.devcycle.com/sdk/server-side-sdks/node/node-openfeature#context-properties-to-devcycleuser

Comment (jonathannorris) [reply lib/Model/DevCycleUser.php:710]: I think we need some form of tests here before we release this. But we can merge this PR ahead of that.

Comment (JamieSinn) [reply lib/Model/DevCycleUser.php:710]: Yeah - this isn't intended to be a one and done - just the first step in a major refactor that needs to also be matched on the test harness

File: lib/OpenFeature/DevCycleProvider.php [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+        return $this->apiClient->variable($user, $flagKey, $defaultValue)->asResolutionDetails();
+    }
+
+    /**
+     * @param string $flagKey
+     * @param array $defaultValue
+     * @param EvaluationContext|null $context
+     * @return ResolutionDetails
+     */
+    public function resolveObjectValue(string $flagKey, array $defaultValue, ?EvaluationContext $context = null): ResolutionDetails
```
Comment (jonathannorris) [inline lib/OpenFeature/DevCycleProvider.php:129]: This will need a bunch of tests to check for all the different JSON restrictions: https://docs.devcycle.com/sdk/server-side-sdks/node/node-openfeature#json-flag-limitations

**Review by jonathannorris — APPROVED** [review]

File: lib/OpenFeature/DevCycleProvider.php (hunk 5 above)
Comment (JamieSinn) [reply lib/OpenFeature/DevCycleProvider.php:129]: This one is a bit tricky - but yes; the handling in PHP is a bit cleaner than other languages thanks to the array concept being basically a dict.
